### PR TITLE
Rfb rust keywords 3195 v1

### PR DESCRIPTION
**Pull request by catenacyber**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/
https://redmine.openinfosecfoundation.org/issues/3195 justify everything

Describe changes:
- rfb: move keywords to pure rust
- rfb: move app-layer to pure rust as well, and convert unit test to SV one

Coming after the MQTT merged PR https://github.com/OISF/suricata/pull/11502, and before other protocols

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1983

cc @satta

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 21588

File: doc/userguide/rules/rfb-keywords.rst [hunk 1]
```diff
@@ -25,9 +25,12 @@ rfb.secresult
 
 Match on the value of the RFB security result, e.g. ``ok``, ``fail``, ``toomany`` or ``unknown``.
 
+rfb.secresult uses an :ref:`unsigned 32-bit integer <rules-integer-keywords>`.
+
 Examples::
 
   rfb.secresult: ok;
+  rfb.secresult: !0;
```
Comment (jufajardini) [inline doc/userguide/rules/rfb-keywords.rst:33]: Considering this is new, I think it'd be worth adding a new rule to https://github.com/OISF/suricata-verify/blob/master/tests/rfb-rules/test.rules to include this format. (I can do that, if the idea is agreed upon :) )

**Comment by jufajardini** [comment]
nit with commit message: s/keywrods/keywords :P

**Comment by jufajardini** [comment]
2nd nit: code looks good, but I think we should have a dedicated ticket for the RFB work, instead of linking the umbrella one.

File: doc/userguide/rules/rfb-keywords.rst (hunk 1 above)
Comment (catenacyber) [reply doc/userguide/rules/rfb-keywords.rst:33]: added in https://github.com/OISF/suricata-verify/pull/1985

**Comment by catenacyber** [comment]
Continued in https://github.com/OISF/suricata/pull/11527

> I think we should have a dedicated ticket for the RFB work, instead of linking the umbrella one.

Do you mean to create 2 tracking tickets ?
- one for moving app-layer-rustprotoX.c to pure rust (like app-layer-rfb.c, but there is also app-layer-ike and other)
- one for moving keywords to rust

+ all the tickets per protocol which need it ?

**Comment by jufajardini** [comment]
> Continued in #11527
> 
> > I think we should have a dedicated ticket for the RFB work, instead of linking the umbrella one.
> 
> Do you mean to create 2 tracking tickets ?
> 
>     * one for moving app-layer-rustprotoX.c to pure rust (like app-layer-rfb.c, but there is also app-layer-ike and other)
> 
>     * one for moving keywords to rust
> 
> 
>     * all the tickets per protocol which need it ?

Maybe just one ticket per protocol to track the changes?

**Pull request closed without merging**